src/Microsoft.OpenApi/Services/OpenApiWalker.cs

```diff
@@ -78,11 +87,15 @@ internal void Walk(OpenApiComponents components)
         /// <param name="paths"></param>
         internal void Walk(OpenApiPaths paths)
         {
+            _visitor.Enter(OpenApiConstants.Paths);
             _visitor.Visit(paths);
-            foreach (var pathItem in paths.Values)
+            foreach (var pathItem in paths)
             {
-                Walk(pathItem);
+                _visitor.Enter(pathItem.Key.Replace("/","~1"));
```

Comment:
>~1" [](start = 57, length = 3)

nit, could you put comment in the code to explain that "~1" is an escape character for "/" in JSON pointer?